server/src/main/java/org/elasticsearch/action/admin/cluster/node/remove/TransportPrevalidateNodeRemovalAction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.node.remove;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.action.admin.cluster.node.remove.NodesRemovalPrevalidation.IsSafe;
+import static org.elasticsearch.action.admin.cluster.node.remove.NodesRemovalPrevalidation.Result;
+
+// TODO: should this instead extend TransportMasterNodeReadAction?
+public class TransportPrevalidateNodeRemovalAction extends TransportAction<PrevalidateNodeRemovalRequest, PrevalidateNodeRemovalResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportPrevalidateNodeRemovalAction.class);
+
+    private final NodeClient client;
+
+    @Inject
+    public TransportPrevalidateNodeRemovalAction(ActionFilters actionFilters, TransportService transportService, NodeClient client) {
+        super(PrevalidateNodeRemovalAction.NAME, actionFilters, transportService.getTaskManager());
+        this.client = client;
+    }
+
+    @Override
+    protected void doExecute(Task task, PrevalidateNodeRemovalRequest request, ActionListener<PrevalidateNodeRemovalResponse> listener) {
+        // TODO: Need to set masterNodeTimeOut?
+        client.admin().cluster().health(new ClusterHealthRequest(), new ActionListener<>() {
```

Comment:
`new ClusterStateHealth(clusterService.state()).getStatus()` is preferable here, avoids the async, avoids another action needing to have circuit breaking disabled and ensures we can use the same `ClusterState` for a full prevalidation as we add more features.